clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsBuilderTest.java

```diff
@@ -72,19 +70,28 @@ public String toString() {
             List<Arguments> values = new ArrayList<>();
             for (int bufferOffset : Arrays.asList(0, 15))
                 for (CompressionType compressionType : CompressionType.values())
-                    values.add(Arguments.of(new Args(bufferOffset, compressionType)));
+                    if (accept(compressionType))
+                        values.add(Arguments.of(new Args(bufferOffset, compressionType)));
             return values.stream();
         }
+        protected boolean accept(CompressionType type) {
+            return true;
+        }
+    }
+
+    private static class NotZstd extends MemoryRecordsBuilderArgumentsProvider {
```

Comment:
I think `NotZstd` is not the best way to capture this. Basically we are excluding the compression types based on the magic version. Is that a better way to describe it?

Reply:
Also, not sure we want to do this inheritance based approach. How about we have the Stream returning part as a method and then you just do the filter on the Stream?

Reply:
@g1geordie It seems you changed the approach according to above comments. I'm +1 to the new provider "NotZstd" as it is a simple solution. In order to make it readable, we can add some comment for "NotZstd" (why we need this provider). Also, the inheritance is unnecessary since it can be replaced by filtered Stream (from a private static method which offer all parameters)

@ijuma WDYT?

Reply:
BTW, ```testAppendedChecksumConsistency``` need to test with zstd. The "magic array" should be changed to magic_2 when input is zstd.

Reply:
What we have here is not exactly what I had intended. I have to take a closer look, but just wanted to share that.

Reply:
Let me explain 

The annotation `@RecordSource` supply valid `( compress , magic )` mapping from magic .  
If minMaigc or maxMaigc is set , it will retrun valid `(compress , magic in range)` and symbolize the test  test in what version more clearly .
 
I don't know this style for both of you is more meaningful or complex.

Reply:
However, the ```@RecordBuilderSource(haveInvalidCompress = true)``` is a inscrutable to me :(

Reply:
@chia7712  
Sorry about my bad english .

In some test  , we want to test unsupported compress .  like  `(compress = ZSTD, magic = V1 ) `
`haveInvalidCompress  = true  `, will include those non valid pair .

Reply:
My point was ```haveInvalidCompress``` is hard to understand which compression is invalid. In current test cases, only zstd is possible to be excluded, right? If so, the name like "NotZstd" is more readable to me.

Reply:
haha . Although I have reverted .
 
`haveInvalidCompress= false`   => (ZSTD , 2)  ...other
`haveInvalidCompress= true`   => (ZSTD , 0) , (ZSTD , 1) ,(ZSTD , 2) ...other

`NotZstd` =>  ...other

Little diff . Never mind me.